blues/moves.py

```diff
@@ -177,70 +178,304 @@ def move(self, context):
-            print('moves and probability list lengths need to match')
-            raise IndexError
-
-
-    def runEngine(self, context):
-        """Selects a random Move object based on its
-        assigned probability and and performs its move() function
-        on a context.
-        Parameters
-        ----------
-        context : openmm.context object
-        OpenMM context whose positions should be moved.
-        """
-        rand_num = np.random.choice(len(self.probs), p=self.probs)
-        try:
-            new_context = self.moves[rand_num].move(context)
-        except Exception as e:
-            #In case the move isn't properly implemented, print out useful info
-            print('Error: move not implemented correctly, printing traceback:')
-            ex_type, ex, tb = sys.exc_info()
-            traceback.print_tb(tb)
-            print(e)
-            raise SystemExit
-
-        return new_context
+        atom_indices = self.atom_indices
+        context = nc_context
+
+        stateinfo = context.getState(True, True, False, True, True, False)
+        oldDartPos = stateinfo.getPositions(asNumpy=True)
+        lig_pos = np.asarray(oldDartPos._value)[self.atom_indices]*unit.nanometers
+        self._findDart(context)
+        center = self.getCenterOfMass(lig_pos, self.masses)
+        selectedboard, changevec = self._calc_from_center(com=center)
+        if selectedboard != None:
+            #TODO just use oldDartPos instead of using temp newDartPos
+            newDartPos = np.copy(oldDartPos)
+            comMove = self._reDart(changevec)
+            vecMove = comMove - center
+            for residue in atom_indices:
+                newDartPos[residue] = newDartPos[residue] + vecMove._value
+            context.setPositions(newDartPos)
+
+            return context
+
+    def _calc_from_center(self, com):
+
+        distList = []
+        diffList = []
+        indexList = []
+        #Find the distances of the COM to each dart, appending
+        #the results to distList
+        for dart in self.dartboard:
+            diff = com - dart
+            dist = np.sqrt(np.sum((diff)*(diff)))*unit.nanometers
+            distList.append(dist)
+            diffList.append(diff)
+        selected = []
+        #Find the dart(s) less than self.dart_size
+        for index, entry in enumerate(distList):
+            if entry <= self.dart_size:
+                selected.append(entry)
+                diff = diffList[index]
+                indexList.append(index)
+        #Dart error checking
+        #to ensure reversibility the COM should only be
+        #within self.dart_size of one dart
+        if len(selected) == 1:
+            return selected[0], diffList[indexList[0]]
+        elif len(selected) == 0:
+            return None, diff
+        elif len(selected) >= 2:
+            #COM should never be within two different darts
+            raise ValueError('sphere size overlap, check darts')
```

Comment:
(Point is: Exceptions should tell the user what has happened and why it has happened, especially when they are exceptions that may reasonably be expected to occur, as this one is!)